src/inmanta/server/services/resourceservice.py

```diff
@@ -227,10 +227,10 @@ async def get_resources_for_agent(
         self, env: data.Environment, agent: str, version: int, sid: uuid.UUID, incremental_deploy: bool
     ) -> Apireturn:
         if not self.agentmanager_service.is_primary(env, sid, agent):
-            return 409, {"message": "This agent is not currently the primary for the endpoint %s (sid: %s)" % (agent, sid)}
+            raise Conflict("This agent is not currently the primary for the endpoint %s (sid: %s)" % (agent, sid))
         if incremental_deploy:
             if version is not None:
-                return 500, {"message": "Cannot request increment for a specific version"}
+                raise Forbidden("Cannot request increment for a specific version")
```

Comment:
500 != forbidden

Reply:
I know, but this should not be a 500 error, right?

Reply:
No it isn't. It's an error caused by a combination of parameter set in the API. It should be a 400. @wouterdb Do you agree?

Reply:
A 400 looks better I agree, here it is https://github.com/inmanta/inmanta/pull/2521/commits/cc22de2ff73581609eb4fc3ce55150fba0a6d50b